### PR TITLE
Support to register a custom plugin

### PR DESCRIPTION
**Pull request by whiskeysierra**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,11 +189,42 @@ on it.
...
+- `OriginalStackTracePlugin`, preserves stack traces when executing requests asynchronously
+- [`FailsafePlugin`](riptide-failsafe), adds retries and circuit breaker support
+- [`TransientFaultPlugin`](riptide-faults), detects transient faults, e.g. network issues
+- [`TimeoutPlugin`](riptide-timeout), applies timeouts to the whole call (including retries, network latency, etc.)
+
+Whenever you encounter the need to perform some repetitive task on the futures returned by a remote call,
+you may consider implementing a custom Plugin for it, e.g.:
+
+```java
+class MetricsPlugin implements Plugin {
```
Comment (lukasniemeier-zalando) [inline README.md]: maybe add a one-liner to this example on how to register such a plugin (for people too lazy to read APIs, not for me...)

File: riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/HttpClientFactoryBean.java [hunk 2]
```diff
@@ -22,7 +23,7 @@
 import static java.lang.String.format;
 import static org.apache.http.conn.ssl.SSLConnectionSocketFactory.getDefaultHostnameVerifier;
 
-class HttpClientFactoryBean implements FactoryBean<HttpClient> {
+class HttpClientFactoryBean extends AbstractFactoryBean<CloseableHttpClient> {
```
Comment (lukasniemeier-zalando) [inline riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/HttpClientFactoryBean.java]: needs a extended unit test now (coverage) :wink:

File: README.md (hunk 1 above)
Comment (whiskeysierra) [reply README.md]: Done, even though it's shown earlier in the readme, in general

File: riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/HttpClientFactoryBean.java (hunk 2 above)
Comment (whiskeysierra) [reply riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/HttpClientFactoryBean.java]: Locally it's covered...

File: README.md (hunk 1 above)
Comment (lukasniemeier-zalando) [reply README.md]: Ah ok missed that, still I like it with it included :wink:

File: riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/HttpClientFactoryBean.java (hunk 2 above)
Comment (whiskeysierra) [reply riptide-spring-boot-starter/src/main/java/org/zalando/riptide/spring/HttpClientFactoryBean.java]: The merge is also fine...

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**